### ISSUE TITLE
[Questions/Issue] Upload, users & issue solving

### ISSUE THREAD

**Issue by Synt3x** [ShaneIsrael/fireshare#109]
Hello,

I'm not sure if I'm missing something but I have two questions.
* 1 Is it possible to add users, so my friend can upload their clip to my fireshare?
* 2 ~Is it possible to upload the video through the web gui? If it's possible I can't seem to find the button.~ Found it, but had to refresh browser after saving "Show Admin Upload Card".
* 3 Is there any debug mode so I can see the http request the service receives?

Thanks in advance.

/Robin

**Comment by ShaneIsrael** [comment]
1. At the moment there isn't a user system in place outside of a single administrator. You could enable public uploads and your friend could upload videos through that. I don't currently have any plans on adding that in as it would require building out a permission system and a lot of UI work. Multiple users was never something that was planned. Sorry. 
3. There isn't a debug mode unfortunately for viewing the internal http traffic. However, if you have Fireshare proxied behind a service like nginx you could simply view those nginx logs.

**Comment by ShaneIsrael** [comment]
Actually I'm wrong about point #2 sort of. By default we already log all the internal traffic. You just need to view the docker logs to see them.

**Comment by Synt3x** [comment]
Thanks for the reply.

I am currently using traefik as an reverse proxy and I can not upload any files. The failing part is on my end because it works without traefik.


I watch the docker logs and it doesn't say much. So I need to keep digging. :)

**Comment by ShaneIsrael** [comment]
Make sure in your traefic fireshare proxy you are setting the

```
client_max_body_size 0;
proxy_read_timeout 999999s;
```

Or whatever traefics settings for those are called. In nginx setting it to
0 allows any size of upload. By default the max body size is only a couple
mbs and most video uploads will be too large and fail so you need to update
that.

The reason it works when you try it from outside of the proxy is because
the Fireshare internal nginx is set to allow any size upload.

**Comment by Synt3x** [comment]
The proxy (Traefik) work with Edge but not Firefox. The fireshare container is just http status 400. But I would like to dump the request! 😅

Need to dig a bit more.

**Comment by ShaneIsrael** [comment]
I would look at the Traefik logs as well, it's likely an issue with your proxy rather than an issue with Fireshare. Unfortunately I don't know anything about Traefik and only really have experience with Nginx.

**Comment by Synt3x** [comment]
Hmm, this is very odd. Is it possible for you to output the how the request looks like in Fireshare?
I can see that the request Traefik forwarding from Edge and Firefox looks more or less the same, but somehow I receive HTTP 400 with Firefox and the transfer cancels.

Works:
* Edge > Fireshare 
* Firefox > Fireshare
* Edge > Traefik > Fireshare

Not working:
* Firefox > Traefik > Fireshare

`"POST /api/upload HTTP/1.1" 400 0 "https://my.domain.com/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:103.0) Gecko/20100101 Firefox/103.0" "172.18.0.3"`

**Comment by ShaneIsrael** [comment]
If you exec into the running Fireshare container you should be able to tail
the nginx logs which are located at /var/log/nginx/access.log

docker exec -it <container I'd> bash

Then just cat or tail the nginx logs file.

**Comment by Synt3x** [comment]
Can't seem to find the issue. Have done a lot of troubleshooting. Can't reproduce it on my laptop either. Friend1 can upload to my fireshare with Firefox and traefik.But my friend2 can't upload to fireshare when traefik is doing the reverse proxy (same as me).

I give up! 😅

**Comment by ShaneIsrael** [comment]
Are they only upload MP4 files? Fireshare only supports a couple file
formats. MP4 is the best to use.

Also check that they are uploaded as h.264 mp4 instead h.265 as that might
also be the issue.

**Comment by Synt3x** [comment]
Yes, the file works. We have tested uploading the files without traefik as reverse proxy and they works.

The problem only occurs for me with firefox and traefik. It's hard to tell when watching the logs but it might be that Firefox is canceling the upload and then Fireshare is returning HTTP 400 because of incomplete upload.

**Comment by ShaneIsrael** [comment]
Yeah the only thing I can say is it's either an issue in the Traefik configuration or an issue with something your friend is doing. 

I primarily use Firefox as my browser and I don't have any issues with uploading to Fireshare through a reverse proxy. The only difference is that I'm using nginx. As long as you've set the appropriate configs in Traefix for allowing uploads (increasing the timeout on uploads and increasing the max file size for uploads) it should just work.

**Comment by Synt3x** [comment]
Traefik should be correct and it works with other services like gitlab, nextcloud and filebrowser. 

The strange part is that it works on my laptop and it's pretty identical as my desktop. 🤨

Sorry for all the stupid questions. I understand that this is like finding a needle in a haystack. I'm really confused why it works on some devices and some it doesn't.

**Comment by ShaneIsrael** [comment]
Well if it's working perfectly fine going directly to the web ui and not through your proxy then the issue doesn't lie with Fireshare, there has to be some outside source causing it whether that be your Traefik config or something else. 

I'm going to close this issue for now. If you eventually figure out what the problem please let me know.

**Comment by Synt3x** [comment]
I just want to let you know that my problem is know solved. I don't know what was causing the issue to happened, but I formatted my computer today and reinstalled the same build of Firefox (and addons) as before, and it know works with Traefik. Before I reinstalled my computer I reinstalled Firefox and it didn't work, so I really don't know what was causing the issue.

I'll let you know if the issue appears again and I notice what may be causing it.
Thanks for all the help!
